src/VisualStudio/Core/Def/Implementation/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -130,25 +129,27 @@ public async Task NavigateToSourceGeneratedFileAsync(SourceGeneratedDocument doc
                 File.WriteAllText(temporaryFilePath, "");
             }
 
-            var openDocumentService = _serviceProvider.GetService<SVsUIShellOpenDocument, IVsUIShellOpenDocument>();
-            var hr = openDocumentService.OpenDocumentViaProject(
-                temporaryFilePath,
-                VSConstants.LOGVIEWID.TextView_guid,
-                out _,
-                out _,
-                out _,
-                out var windowFrame);
-
-            if (ErrorHandler.Succeeded(hr) && windowFrame != null)
+            return async cancellationToken =>
             {
-                windowFrame.Show();
-            }
+                await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+                var openDocumentService = _serviceProvider.GetService<SVsUIShellOpenDocument, IVsUIShellOpenDocument>();
```

Comment:
could potentially do this on the BG as well.  but i think ti's highly unlikely that will help anything.  this is MAS.  we've already loaded docs, so this service is already goign to be avialable.

Reply:
Agreed, seems unlikely for that to be expensive. :smile: